File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task85_CLAUDE_claude-3-5-sonnet-20240620.cpp

#include <iostream>
#include <string>
#include <random>
#include <chrono>

class RandomStringGenerator {
private:
    static const std::string ALPHABET;
    static std::mt19937 rng;

public:
    static std::string generateRandomString(int length) {
        std::uniform_int_distribution<int> dist(0, ALPHABET.length() - 1);
        std::string result;
        result.reserve(length);

        for (int i = 0; i < length; ++i) {
            result += ALPHABET[dist(rng)];
        }

        return result;
    }
};

const std::string RandomStringGenerator::ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
std::mt19937 RandomStringGenerator::rng(std::chrono::steady_clock::now().time_since_epoch().count());

int main() {
    int length = 10;
    std::string randomString = RandomStringGenerator::generateRandomString(length);
    std::cout << "Random string: " << randomString << std::endl;
    return 0;
}

File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task85_CLAUDE_claude-3-5-sonnet-20240620.java

import java.security.SecureRandom;

public class RandomStringGenerator {
    private static final String ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
    private static final SecureRandom RANDOM = new SecureRandom();

    public static String generateRandomString(int length) {
        StringBuilder sb = new StringBuilder(length);
        for (int i = 0; i < length; i++) {
            int index = RANDOM.nextInt(ALPHABET.length());
            sb.append(ALPHABET.charAt(index));
        }
        return sb.toString();
    }

    public static void main(String[] args) {
        int length = 10;
        String randomString = generateRandomString(length);
        System.out.println("Random string: " + randomString);
    }
}
